js/pages/tola_management_pages/country/__tests__/countryAdminPage.test.js

```diff
@@ -1,293 +1,34 @@
-            {id: 200, name: "Program 1", checked: false}
-
-        ];
-        let wrapper = mount(<RetroProgramCheckBoxWrapper programs={programsForRetro} programsExpanded={true}/>);
-        let checkBoxList = wrapper.find(CheckBoxList);
-        expect(checkBoxList.props().checkBoxOptions).toEqual(programsForRetro.reverse());
-    });
-
-    it("doesn't include programs in saved data when none are checked", () => {
-        const programsForRetro = [
-            {id: 200, name: "Program 1", checked: false},
-            {id: 201, name: "Program 2", checked: false}
-        ];
-        const disaggregation = {...store.editing_disaggregations_data[0]};
-        disaggregation.id = "new";
-        disaggregation.selected_by_default = true;
-        let wrapper = shallow(<DisaggregationType
-            programs={programsForRetro}
-            disaggregation={disaggregation}
-            saveDisaggregation={(data) => saveDisaggregation(data)}
-        />);
-        wrapper.setState({selected_by_default: true});
-        wrapper.instance().programsForRetro = programsForRetro;
-        wrapper.instance().save();
-        expect(saveDisaggregation.mock.calls[0][0]).toEqual(disaggregation);
-    });
-
-    it("includes programs in saved data when one or more are checked", () => {
-        const programsForRetro = [
-            {id: 200, name: "Program 1", checked: false},
-            {id: 201, name: "Program 2", checked: true}
-        ]
-        let disaggregation = {...store.editing_disaggregations_data[0]}
-        disaggregation.id = "new"
-        disaggregation.selected_by_default = true;
-        let wrapper = shallow(<DisaggregationType
-            programs={programsForRetro}
-            disaggregation={disaggregation}
-            saveDisaggregation={(data) => saveDisaggregation(data)}
-        />);
-        wrapper.setState({selected_by_default: true});
-        wrapper.instance().programsForRetro = programsForRetro;
-        wrapper.instance().save();
-        disaggregation.retroPrograms = [201];
-        expect(saveDisaggregation.mock.calls[1][0]).toEqual(disaggregation);
-    });
-});
+    store.editing_objectives_data = strategicObjectiveData;
+
+    it("clears stale country values upon country creation", () => {
+        store.countries = [...store.allCountries];
+        store.editing_errors = {'test1': 'error1'};
+        store.editing_disaggregations_errors = {'test2': 'error2'};
+        store.editing_target = 100;
+
+        expect(store.editing_objectives_data.length).toBeGreaterThan(0);
+        expect(store.editing_disaggregations_data.length).toBeGreaterThan(0);
+        expect(Object.keys(store.editing_errors).length).toBeGreaterThan(0);
+        expect(Object.keys(store.editing_disaggregations_errors).length).toBeGreaterThan(0);
+        store.addCountry();
+        expect(store.editing_objectives_data.length).toEqual(0);
+        expect(store.editing_objectives_data.length).toEqual(0);
```

Comment:
Theres a duplicate test here, I think you meant editing_disaggregation_data_length.

Reply:
Ah, good catch.  Fixed.